test/src/tests/widgets/test_label.py

```diff
@@ -0,0 +1,44 @@
+from pytest import approx, fixture
+
+import toga
+
+from ..test_data import COLORS, TEXTS
+from ..test_utils import set_get
+
+
+@fixture
+async def new_widget():
+    return toga.Label("")
+
+
+async def test_text(widget, probe):
+    for text in TEXTS:
+        set_get(widget, "text", text)
+        assert probe.text == text
+
+
+async def test_color(widget, probe):
+    for color in COLORS:
+        widget.style.color = color
+        for component in ["r", "g", "b"]:
+            assert getattr(probe.color, component) == getattr(color, component)
+        assert probe.color.a == approx(color.a, abs=(1 / 256))
+
+
+async def test_multiline(widget, probe):
+    def make_lines(n):
+        return "\n".join(f"line{i}" for i in range(n))
+
+    widget.text = make_lines(1)
+    line_height = probe.height
+
+    widget.text = make_lines(2)
+    assert probe.height == approx(line_height * 2, rel=0.1)
+    line_spacing = probe.height - (line_height * 2)
```

Comment:
This test currently fails on Windows, probably because of #1289, and maybe also because we need an `await` to allow the size change to take effect, as we will on Android.

Reply:
In order to keep this PR manageable, I won't include any fixes for bugs in Toga itself. If I come across any more while adding tests, I'll just skip the test with a comment.